### PR TITLE
Icu 14400 add a section for version info cli and dc

### PR DESCRIPTION
**Pull request by DhariniJeeva**
# Description
https://hashicorp.atlassian.net/browse/ICU-14400
This PR adds application information section on the user settings page in DC

## Screenshots (if appropriate)
 
 
Screenshot 2024-07-30 at 8 54 17 PM

with no daemon:
 
Screenshot 2024-07-30 at 8 48 09 PM
 

## Checklist

- [x] I have added before and after screenshots for UI changes
- ~[ ] I have added JSON response output for API changes~
- ~[ ] I have added steps to reproduce and test for bug fixes in the description~
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/desktop/app/components/settings-card/index.hbs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+<Hds::Card::Container @level='mid' @hasBorder={{true}} class='settings-card'>
+
+  <div class='header-and-icon'>
+    <Hds::IconTile @icon={{@icon}} @size='small' />
+    <Hds::Text::Display @tag='h4'>
+      {{@header}}
+    </Hds::Text::Display>
+
+    {{#if @action}}
```
Comment (DhariniJeeva) [inline ui/desktop/app/components/settings-card/index.hbs:14]: this is not used right now, it is for the action buttons that we will have in the next tickets

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs [hunk 2]
```diff
@@ -3,4 +3,61 @@
...
+      <div>
+        <Hds::Text::Display @tag='h5'>
+          {{t 'titles.cli'}}
+        </Hds::Text::Display>
+        <Hds::Text::Body @tag='p' @size='300'>
+          {{#if @model.cliExists}}
+            {{t 'states.installed'}}
+          {{else}}
+            {{t 'states.not-available'}}
+          {{/if}}
```
Comment (ZedLi) [inline ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:26]: I don't think this is very useful, since it should never be the case where the CLI isn't installed. We can show the `not-available` message if for some reason it isn't since that should be an exceptional case, but shouldn't we display the CLI version here? And the Version on line 14 be the desktop client version?

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs [hunk 3]
```diff
@@ -3,4 +3,61 @@
...
+      <div>
+        <Hds::Text::Display @tag='h5'>
+          {{t 'titles.cache-daemon'}}
+        </Hds::Text::Display>
+        <Hds::Text::Body @tag='p' @size='300'>
+          {{#if @model.getCacheDaemonVersion}}
+            {{@model.getCacheDaemonVersion}}
+            <Hds::Badge
+              @text={{t 'states.running'}}
+              @icon='loading'
```
Comment (ZedLi) [inline ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:38]: Shouldn't this be a `running` icon instead of loading?

File: ui/desktop/electron-app/src/ipc/handlers.js [hunk 4]
```diff
@@ -194,6 +194,26 @@ handle('isClientAgentRunning', async () => {
...
 
+/**
+ * Gets boundary cli version
+ */
+handle('getCliVersion', async () => boundaryCli.version());
+
+/**
+ * Gets Cache Daemon version
+ */
+handle('getCacheDaemonVersion', async () => {
```
Comment (ZedLi) [inline ui/desktop/electron-app/src/ipc/handlers.js:205]: Shouldn't we be returning the full status instead of just the version here and have the app handle parsing out the version? We likely will need more later

File: ui/desktop/app/controllers/scopes/scope/projects/settings/index.js [hunk 5]
```diff
@@ -4,5 +4,60 @@
...
+  get themes() {
+    return THEMES;
+  }
+
+  /**
+   * Applies the specified color theme to the root ember element.
+   * @param {string} theme - "light", "dark", or nullish (system default)
+   */
+  @action
+  async toggleTheme({ target: { value: theme } }) {
```
Comment (ZedLi) [inline ui/desktop/app/controllers/scopes/scope/projects/settings/index.js:43]: Should we be removing the theme selector from the global dropdown now?

Comment (DhariniJeeva) [reply ui/desktop/app/controllers/scopes/scope/projects/settings/index.js:43]: yup, there was a note on figma that we should remove the global theme selector, I added a separate task for that

Comment (calcaide) [reply ui/desktop/app/controllers/scopes/scope/projects/settings/index.js:43]: just to clarify, is this `toggleTheme` function JUST for the theme selector on the global dropdown?

File: ui/desktop/app/components/settings-card/index.hbs [hunk 6]
```diff
@@ -0,0 +1,36 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+}}
+
+<Hds::Card::Container @level='mid' @hasBorder={{true}} class='settings-card'>
```
Comment (calcaide) [inline ui/desktop/app/components/settings-card/index.hbs:6]: After reviewing figma designs, I notice:
- All the cards have at least header and body/content.
- The headers in all the cards, at least have icon + text. Some have status and/or actions.
- The bodies/content seems to be a grid with a variety of content: text, components, etc

I will suggest within <hds::card>
We have 2 divs: `.header`, `.body/content` and try to follow and reproduce this pattern within all the cards, this will facilitate re use and clarify for the future CSS.

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs (hunk 2 above)
Comment (calcaide) [reply ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:26]: Indeed, I just randomly tested (for something else) last week to run the DC with no CLI and it's useless and it breaks, so I do strongly believe no one will be able to open DC, connect to a controller and show settings page unless there is a CLI present on the system.

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs [hunk 7]
```diff
@@ -0,0 +1,63 @@
+{{!
+  Copyright (c) HashiCorp, Inc.
+  SPDX-License-Identifier: BUSL-1.1
+}}
+
+<SettingsCard @header={{t 'titles.application'}} @icon='boundary'>
+  <div class='content'>
+    <div class='section'>
```
Comment (calcaide) [inline ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:8]: Can we ellaborate on the names we are assigning to clases here? `content` is clear, but then `section` and after `theme`? seems to me these is quite ad-hoc, but this is a pattern that we will reproduce in other settings cards, can we look at something more generic?

File: ui/desktop/app/styles/app.scss [hunk 8]
```diff
@@ -662,3 +662,29 @@ $chevron-down: url("data:image/svg+xml;utf-8,<svg viewBox='0 0 24 24' fill='%231
   gap: 0.5rem;
   width: inherit;
 }
+
+// Settings
+.settings-card {
```
Comment (calcaide) [inline ui/desktop/app/styles/app.scss:667]: I am not sure we need an specific class. What do you think on giving an id to the whole settings page? then we can select #settings > hds card. That way the CSS will apply to every hds card element within settings, which I think is what we want. Thoughts?

File: ui/desktop/electron-app/src/ipc/handlers.js (hunk 4 above)
Comment (calcaide) [reply ui/desktop/electron-app/src/ipc/handlers.js:205]: Agreed, this could be `getCacheStatus` and then the model of an specific domain can parse out what's not need.

**Review by calcaide — CHANGES_REQUESTED** [review]
Looking good!! I left some comments

File: ui/desktop/app/styles/app.scss (hunk 8 above)
Comment (DhariniJeeva) [reply ui/desktop/app/styles/app.scss:667]: this needs to be a `classname`, because this component will be re-used in building out the rest of the cards.

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs (hunk 7 above)
Comment (DhariniJeeva) [reply ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:8]: agreed it could be better, I updated this

File: ui/desktop/app/components/settings-card/index.hbs (hunk 6 above)
Comment (DhariniJeeva) [reply ui/desktop/app/components/settings-card/index.hbs:6]: I have a header div block and yield the body/content from the consumers. Added a comment for more clarity

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs (hunk 2 above)
Comment (DhariniJeeva) [reply ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs]: removed this

File: ui/desktop/electron-app/src/ipc/handlers.js (hunk 4 above)
Comment (DhariniJeeva) [reply ui/desktop/electron-app/src/ipc/handlers.js]: thanks, updated to receive full status from cache

File: ui/desktop/app/controllers/scopes/scope/projects/settings/index.js (hunk 5 above)
Comment (DhariniJeeva) [reply ui/desktop/app/controllers/scopes/scope/projects/settings/index.js:43]: 🤔 I don't think I follow, can you explain? 

we won't have a theme selector on the global dropdown going forward

Comment (calcaide) [reply ui/desktop/app/controllers/scopes/scope/projects/settings/index.js:43]: I was asking if `toggleTheme` functionality does toggle the theme only within the dropdown that is rendered within settings or if is to toggle the theme within the global dropdown?

File: ui/desktop/app/styles/app.scss (hunk 8 above)
Comment (calcaide) [reply ui/desktop/app/styles/app.scss:667]: What I mean is if the main element or [the div that wraps the page body](https://github.com/hashicorp/boundary-ui/pull/2409/files#diff-23bd88f09a10c21b4b497fa14821c75c0276e1e2553634c6dee953e4a78804deR6) we add an id such as `#settings` or `#settings-section` then, you do not need to add a classname to every card we will add, as we do here. You can reference this on the css as `#settings > hds_card` (or the default class HDS generated for hds_card). And those styles will apply to all the HDS cards that are within settings section, without need for `classname`

All good with the current implementation, just anticipating this section will be several cards, I though we could reduce classname usage.

File: ui/desktop/app/controllers/scopes/scope/projects/settings/index.js (hunk 5 above)
Comment (DhariniJeeva) [reply ui/desktop/app/controllers/scopes/scope/projects/settings/index.js:43]: only within the settings page, I have another ticket that gets rid off the global dropdown

File: ui/desktop/electron-app/src/ipc/handlers.js [hunk 9]
```diff
@@ -194,6 +194,33 @@ handle('isClientAgentRunning', async () => {
...
+});
+
+/**
+ * Gets Cache Daemon version
+ */
+handle('getCacheDaemonStatus', async () => {
+  let cache = '';
+  try {
+    cache = await cacheDaemonManager.status();
+  } catch (e) {
```
Comment (ZedLi) [inline ui/desktop/electron-app/src/ipc/handlers.js:217]: How come we're catching the error here and then just returning it?

File: ui/desktop/electron-app/src/ipc/handlers.js [hunk 10]
```diff
@@ -194,6 +194,33 @@ handle('isClientAgentRunning', async () => {
...
+/**
+ * Returns desktop version
+ */
+handle('getDesktopVersion', async () => {
+  const desktopVersion = app.getVersion();
+  return desktopVersion;
+});
+
+/**
+ * Gets Cache Daemon version
```
Comment (ZedLi) [inline ui/desktop/electron-app/src/ipc/handlers.js:211]:
```suggestion
 * Returns the Cache daemon status
```

File: ui/desktop/electron-app/src/ipc/handlers.js [hunk 11]
```diff
@@ -194,6 +194,33 @@ handle('isClientAgentRunning', async () => {
...
+ */
+handle('getCliVersion', async () => boundaryCli.version());
+
+/**
+ * Returns desktop version
+ */
+handle('getDesktopVersion', async () => {
+  const desktopVersion = app.getVersion();
+  return desktopVersion;
+});
```
Comment (ZedLi) [inline ui/desktop/electron-app/src/ipc/handlers.js:208]:
```suggestion
handle('getDesktopVersion', async () =>  app.getVersion());
```

Does `app.GetVersion` pull the correct version? We normally pull it from our generated config and the docs says it pulls it from the `package.json` which I don't believe we update.

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs [hunk 12]
```diff
@@ -3,4 +3,54 @@
...
+        </Hds::Text::Body>
+      </div>
+      <div>
+        <Hds::Text::Display>
+          {{t 'titles.cache-daemon'}}
+        </Hds::Text::Display>
+        <Hds::Text::Body @tag='p' @weight='medium'>
+          {{#if @model.formattedCacheVersion}}
+            {{@model.formattedCacheVersion}}
+            <Hds::Badge
```
Comment (ZedLi) [inline ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:32]: Should we display a not running with an icon as well if we don't get the version?

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs [hunk 13]
```diff
@@ -3,4 +3,54 @@
...
+        <Hds::Text::Display>
+          {{t 'titles.version'}}
+        </Hds::Text::Display>
+        <Hds::Text::Body @tag='p' @weight='medium'>
+          {{@model.desktopVersion}}
+        </Hds::Text::Body>
+      </div>
+      <div>
+        <Hds::Text::Display>
+          {{t 'titles.cli'}}
```
Comment (ZedLi) [inline ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:19]: Should this say CLI version?

File: ui/desktop/electron-app/src/ipc/handlers.js (hunk 11 above)
Comment (DhariniJeeva) [reply ui/desktop/electron-app/src/ipc/handlers.js:208]: yeah, I wasn't sure which one to use, thought either would work, but good to know we don't keep track of the app one. I will switch this to pull from config

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs (hunk 13 above)
Comment (DhariniJeeva) [reply ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:19]: [figma](https://www.figma.com/design/28zD8tKqa9yUQVO13THm77/Transparent-Sessions-%26-Aliases?node-id=4623-123988&t=Eub6bjk8XC75GRHr-0) says CLI and cache daemon.  

@joshuaogle thoughts?  [this would be the headers in the application info section]

File: ui/desktop/app/routes/scopes/scope/projects/settings/index.js [hunk 14]
```diff
@@ -4,5 +4,21 @@
...
+import { inject as service } from '@ember/service';
 
-export default class ScopesScopeProjectsSettingsIndexRoute extends Route {}
+export default class ScopesScopeProjectsSettingsIndexRoute extends Route {
+  @service ipc;
+
+  async model() {
+    const { versionNumber: cliVersion } =
+      await this.ipc.invoke('getCliVersion');
+    const getCacheDaemon = await this.ipc.invoke('getCacheDaemonStatus');
```
[the remark this answers is not in the document]
Comment (DhariniJeeva) [reply ui/desktop/app/routes/scopes/scope/projects/settings/index.js]: updated!

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs (hunk 12 above)
Comment (DhariniJeeva) [reply ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:32]: updated to show not running and an icon for that

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs (hunk 13 above)
Comment (DhariniJeeva) [reply ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:19]: discussed offline and updated based on design feedback to use prefixes

File: ui/desktop/app/routes/scopes/scope/projects/settings/index.js [hunk 15]
```diff
@@ -4,5 +4,21 @@
...
+    const { version } = getCacheDaemon;
+    // Format the version
+
+    const formattedCacheVersion = version && /(\d+\.\d+\.\d+)/.exec(version)[0];
```
[the remark this answers is not in the document]
Comment (DhariniJeeva) [reply ui/desktop/app/routes/scopes/scope/projects/settings/index.js]: you are right, I did not need that. I was using that in case I did not have a version, but now modified the model to handle the case so it is not needed. thanks for the catch

File: ui/desktop/app/styles/app.scss (hunk 8 above)
Comment (DhariniJeeva) [reply ui/desktop/app/styles/app.scss:667]: agreed, I removed the classname from here and moved it to the wrapper. since we don't use `id` in our codebase for style, I kept it as `classname`. But I'm not opposed to using ID and this could be a team discussion on what the team prefers. adding to tech time

File: ui/desktop/app/routes/scopes/scope/projects/settings/index.js [hunk 16]
```diff
@@ -4,5 +4,39 @@
...
+    formattedDesktopVersion = `v${desktopVersion}`;
+    try {
+      cacheDaemonStatus = await this.ipc.invoke('cacheDaemonStatus');
+    } catch (e) {
+      __electronLog?.warn('cache daemon is not running', e.message);
+      return {
+        formattedCliVersion,
+        formattedCacheVersion,
+        formattedDesktopVersion,
+        warning: true,
```
Comment (DhariniJeeva) [inline ui/desktop/app/routes/scopes/scope/projects/settings/index.js:31]: I'm using this to show the warning alert in the UI, open to feedback on better ways to achieve this

File: ui/desktop/app/styles/app.scss (hunk 8 above)
Comment (calcaide) [reply ui/desktop/app/styles/app.scss:667]: Awesome @DhariniJeeva thanks for the clarification and bringing this topic to tech time 😃

**Review by calcaide — APPROVED** [review]

File: ui/desktop/app/routes/scopes/scope/projects/settings/index.js (hunk 16 above)
Comment (ZedLi) [reply ui/desktop/app/routes/scopes/scope/projects/settings/index.js:31]: It might be better to return the actual error so we can show any error messages?

Comment (DhariniJeeva) [reply ui/desktop/app/routes/scopes/scope/projects/settings/index.js]: good call, I'm returning the err obj, we will display the message in the UI if there's any, let me know what you think

File: ui/desktop/app/routes/scopes/scope/projects/settings/index.js [hunk 17]
```diff
@@ -4,5 +4,39 @@
...
+    const { versionNumber: cliVersion } =
+      await this.ipc.invoke('getCliVersion');
+    formattedCliVersion = `v${cliVersion}`;
+
+    const { desktopVersion } = await this.ipc.invoke('getDesktopVersion');
+    formattedDesktopVersion = `v${desktopVersion}`;
+    try {
+      cacheDaemonStatus = await this.ipc.invoke('cacheDaemonStatus');
+    } catch (e) {
+      __electronLog?.warn('cache daemon is not running', e.message);
```
Comment (ZedLi) [inline ui/desktop/app/routes/scopes/scope/projects/settings/index.js:26]: I don't think this is strictly needed since we already have a log statement if the status fails in electron [here](https://github.com/hashicorp/boundary-ui/blob/10c128ee3baf695e350679f82122ef0cfb8cdd8a/ui/desktop/electron-app/src/services/cache-daemon-manager.js#L40)

File: ui/desktop/app/routes/scopes/scope/projects/settings/index.js [hunk 18]
```diff
@@ -4,5 +4,39 @@
...
+      formattedCliVersion,
+      formattedDesktopVersion;
+    const { versionNumber: cliVersion } =
+      await this.ipc.invoke('getCliVersion');
+    formattedCliVersion = `v${cliVersion}`;
+
+    const { desktopVersion } = await this.ipc.invoke('getDesktopVersion');
+    formattedDesktopVersion = `v${desktopVersion}`;
+    try {
+      cacheDaemonStatus = await this.ipc.invoke('cacheDaemonStatus');
```
Comment (ZedLi) [inline ui/desktop/app/routes/scopes/scope/projects/settings/index.js:24]: We can still get errors as part of a successful response like this but we can handle this in another ticket if you'd like.

```{
  "status_code": 200,
  "item": {
    "uptime": 2100252029000,
    "socket_address": "/Users/zhiheli/.boundary/socket/daemon.sock",
    "log_location": "/Users/zhiheli/.boundary/cache.log",
    "version": "Boundary v0.17.0 (126c4f55a006525ab4574742c517e7b3b76e0336)",
    "users": [
      {
        "id": "u_1234567890",
        "boundary_instance": {
          "address": "http://localhost:9200",
          "cache_support": "Supported",
          "last_support_check": 20486662000
        },
        "auth_tokens": [
          {
            "id": "at_zFm4apTRvb"
          },
          {
            "id": "at_9DZX9QLWrq",
            "keyringless_references": 1
          }
        ],
        "resources": [
          {
            "name": "resolvable-alias",
            "count": 3,
            "last_error": {
              "error": "cache.(Repository).refreshResolvableAliases: for token \"at_9DZX9QLWrq\": cache.defaultResolvableAliasFunc: unknown, unknown: error #0: ",
              "last_returned": 21007917000
            },
            "refresh_token": {
              "age": 328362859000,
              "last_used": 295585859000
            }
          },
          {
            "name": "target",
            "count": 2,
            "refresh_token": {
              "age": 22450018000,
              "last_used": 20487018000
            }
          },
          {
            "name": "session",
            "count": 0
          }
        ]
      }
    ]
  }
}
```

File: ui/desktop/tests/acceptance/projects/settings/index-test.js [hunk 19]
```diff
@@ -9,10 +9,12 @@ import { setupApplicationTest } from 'ember-qunit';
 import setupMirage from 'ember-cli-mirage/test-support/setup-mirage';
 import { authenticateSession } from 'ember-simple-auth/test-support';
 import WindowMockIPC from '../../../helpers/window-mock-ipc';
+import setupStubs from 'api/test-support/handlers/cache-daemon-search';
 
 module('Acceptance | projects | settings | index', function (hooks) {
```
Comment (ZedLi) [inline ui/desktop/tests/acceptance/projects/settings/index-test.js:14]: Should we have tests for the color theme picker here now instead? Maybe we should have the moving of the theme picker in a different PR completely

File: ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs [hunk 20]
```diff
@@ -3,4 +3,73 @@
   SPDX-License-Identifier: BUSL-1.1
 }}
 
-{{outlet}}
\ No newline at end of file
+<SettingsCard @header={{t 'titles.application'}} @icon='boundary'>
+  <div class='content'>
```
Comment (ZedLi) [inline ui/desktop/app/templates/scopes/scope/projects/settings/index.hbs:7]: Should this div be part of the generic component?

File: ui/desktop/app/components/settings-card/index.hbs [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+
+  <div class='header-and-icon'>
+    <Hds::IconTile @icon={{@icon}} @size='small' />
+    <Hds::Text::Display @tag='h4'>
+      {{@header}}
+    </Hds::Text::Display>
+
+    {{#if @action}}
+      <div class='action'>
+        {{yield}}
```
Comment (ZedLi) [inline ui/desktop/app/components/settings-card/index.hbs:16]: How's this work with the yield on line 33 as well?

File: ui/desktop/app/styles/app.scss [hunk 22]
```diff
@@ -662,3 +662,31 @@ $chevron-down: url("data:image/svg+xml;utf-8,<svg viewBox='0 0 24 24' fill='%231
...
+  .content {
+    padding-left: 2.2rem;
+
+    .hds-form-field--layout-vertical {
+      margin-top: 2rem;
+    }
+  }
+
+  .app-information {
+    display: grid;
```
Comment (ZedLi) [inline ui/desktop/app/styles/app.scss:687]: Should this be a grid specifically for 3 columns? I'm wondering if it makes sense to use flex instead? How were you planning on doing the other cards that only have 1 or 2 columns?

File: ui/desktop/app/styles/app.scss [hunk 23]
```diff
@@ -662,3 +662,31 @@ $chevron-down: url("data:image/svg+xml;utf-8,<svg viewBox='0 0 24 24' fill='%231
...
+  .header-and-icon {
+    display: flex;
+    align-items: center;
+    gap: 0.5rem;
+  }
+
+  .content {
+    padding-left: 2.2rem;
+
+    .hds-form-field--layout-vertical {
```
Comment (ZedLi) [inline ui/desktop/app/styles/app.scss:681]: Is this the right selector? It seems like we'll need margin for multiple things in different cards and this is very specific to the theme selector

File: ui/desktop/tests/acceptance/projects/settings/index-test.js (hunk 19 above)
Comment (DhariniJeeva) [reply ui/desktop/tests/acceptance/projects/settings/index-test.js:14]: moved it outside this PR as the tests might need some refactors to follow our most recent patterns

File: ui/desktop/app/components/settings-card/index.hbs (hunk 21 above)
Comment (DhariniJeeva) [reply ui/desktop/app/components/settings-card/index.hbs]: changed it to `@action` as in the next tickets, we will add action button in this block

File: ui/desktop/app/styles/app.scss (hunk 22 above)
Comment (DhariniJeeva) [reply ui/desktop/app/styles/app.scss]: you are right, grid won't work in that case, changed to flex

File: ui/desktop/app/routes/scopes/scope/projects/settings/index.js (hunk 18 above)
Comment (DhariniJeeva) [reply ui/desktop/app/routes/scopes/scope/projects/settings/index.js:24]: yeah I'll make it a seprate ticket

File: ui/desktop/app/components/settings-card/index.hbs (hunk 21 above)
Comment (ZedLi) [reply ui/desktop/app/components/settings-card/index.hbs]: Just want to understand, wouldn't we still need to yield content? What would we be passing to `@action`?

File: ui/desktop/app/routes/scopes/scope/projects/settings/index.js (hunk 18 above)
Comment (ZedLi) [reply ui/desktop/app/routes/scopes/scope/projects/settings/index.js:23]: Yeah it is, it's still a successful response and isn't blocking, errors can happen on any individual resource

File: ui/desktop/app/components/settings-card/index.hbs (hunk 21 above)
Comment (DhariniJeeva) [reply ui/desktop/app/components/settings-card/index.hbs]: hmm, I didn't spend much time thinking this through, my initial thought was to pass in actions(signout/pause/restart) but that wouldn't work if there is more than one. now I'm thinking of passing in a named block to `@actions`?

Comment (ZedLi) [reply ui/desktop/app/components/settings-card/index.hbs]: Yup makes sense, I think we'd have to do named blocks in this component

Comment (DhariniJeeva) [reply ui/desktop/app/components/settings-card/index.hbs]: updated!

**Review by ZedLi — APPROVED** [review]
Looks great, thanks for getting things setup!
